fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -171,10 +172,19 @@ public Response head() {
 
         addResourceHttpHeaders(resource());
 
-        final Response.ResponseBuilder builder = ok();
+        Response.ResponseBuilder builder = ok();
 
         if (resource() instanceof FedoraBinary) {
-            builder.type(((FedoraBinary) resource()).getMimeType());
+            final MediaType mediaType = MediaType.valueOf(((FedoraBinary) resource()).getMimeType());
+
+            if (MESSAGE_EXTERNAL_BODY.isCompatible(mediaType) && mediaType.getParameters().containsKey(
```

Comment:
This is kinda long. It might be nice to factor it out into `boolean isExternal(...)` or something.